internal/rpc/jsonrpc/methods.go

```diff
@@ -5247,20 +5247,24 @@ func (s *Server) walletInfo(ctx context.Context, icmd interface{}) (interface{},
 		return nil, errUnloadedWallet
 	}
 
-	n, err := w.NetworkBackend()
-	connected := err == nil
-	_, spv := n.(*spv.Syncer)
-	if connected {
-		if rpc, ok := n.(*dcrd.RPC); ok {
-			err := rpc.Call(ctx, "ping", nil)
-			if ctx.Err() != nil {
-				return nil, ctx.Err()
-			}
-			if err != nil {
-				log.Warnf("Ping failed on connected daemon client: %v", err)
-				connected = false
-			}
+	var connected, spvMode bool
+	switch n, _ := w.NetworkBackend(); rpc := n.(type) {
+	case *spv.Syncer:
+		spvMode = true
+		connected = len(rpc.GetRemotePeers()) > 0
+	case *dcrd.RPC:
+		err := rpc.Call(ctx, "ping", nil)
+		if ctx.Err() != nil {
+			return nil, ctx.Err()
 		}
+		if err != nil {
+			log.Warnf("Ping failed on connected daemon client: %v", err)
+		} else {
+			connected = true
+		}
+	case nil:
+	default:
+		log.Warnf("walletInfo - invalid network backend (%T).", n)
```

Comment:
@jrick I think you have something else in mind for these two cases.  return a non-nil error with default?

Reply:
Yeah, default case can return an error.  I don't think the nil case should though.  The program is working correctly but we just don't know what syncer is being used, perhaps because one has not yet connected.